src/services/getResolvedRouteForUrl.ts

```diff
@@ -2,14 +2,21 @@ import { readonly } from 'vue'
 import { createMaybeRelativeUrl } from '@/services/createMaybeRelativeUrl'
 import { createResolvedRouteQuery } from '@/services/createResolvedRouteQuery'
 import { getRouteParamValues, routeParamsAreValid } from '@/services/paramValidation'
-import { routePathMatches, routeQueryMatches } from '@/services/routeMatchRegexRules'
+import { isNamedRoute, routeHostMatches, routePathMatches, routeQueryMatches } from '@/services/routeMatchRules'
 import { getRouteScoreSortMethod } from '@/services/routeMatchScore'
 import { ResolvedRoute } from '@/types/resolved'
 import { Routes } from '@/types/route'
 import { RouteMatchRule } from '@/types/routeMatchRule'
 
+const rules: RouteMatchRule[] = [
+  isNamedRoute,
+  routeHostMatches,
+  routePathMatches,
+  routeQueryMatches,
+  routeParamsAreValid,
+]
```

Comment:
We were creating this const every time `getResolvedRouteForUrl` was called. Pulling it out into its own static const.